Publish Code Engine/Paketo post (#2746)
OpenLiberty/blogs: posts/2022-11-14-using-codeengine-with-paketo-buildpack-for-liberty.adoc

---
layout: post
title: "Deploy Java applications to IBM Cloud Code Engine using the Paketo Buildpack for Liberty"
# Do NOT change the categories section
categories: blog
author_picture: https://avatars.githubusercontent.com/u/29545669?v=4
author_github: https://github.com/kevin-ortega
seo-title:  Deploy Java applications to IBM Cloud Code Engine (ICCE) using the Paketo Buildpack for Liberty - OpenLiberty.io
seo-description: Learn to deploy your applications to ICCE easily with the Paketo Buildpack for Liberty
blog_description: "Learn to deploy your applications to ICCE easily with the Paketo Buildpack for Liberty"

> open-graph-image: https://openliberty.io/img/blog/code_engine_build_step_details.png

Reviewer: My suggestion next time is to try using a relative path for the `open-graph-image` value. As of right now, the image used here is not on `openliberty.io` yet.

I usually will test the SEO fields (like the `og:image`) in Slack to see the metadata that gets pulled into preview cards. Here is an example of me DMing myself the link to test the Slack's URL preview cards. Notice an image is missing.
<img width="1044" alt="image" src="https://user-images.githubusercontent.com/31117513/201743393-7df75c0a-d709-44e6-aadf-0d95ccd1c1e1.png">

Reviewer: Woops, I did not give a suggestion on a relative path. Here is what I would of tried `open-graph-image: /img/blog/code_engine_build_step_details.png`.

Reviewer 2: from my recollection, relative paths use to not work for this.... but maybe things have changed or I'm misremembering

> open-graph-image: https://openliberty.io/img/blog/code_engine_build_step_details.png

Reviewer 2: I would have suggested using [code_engine_build_strategy.png](https://github.com/OpenLiberty/blogs/pull/2746/files#diff-f654c4f7cbedd759fea54bf522d90fa0d1504ad6c068c5055c83e914101ec2f0)  instead as it seems more interesting

open-graph-image-alt: screen capture of Code Engine build details
---
= Deploy Java applications to IBM Cloud Code Engine using the Paketo Buildpack for Liberty
Kevin Ortega <https://github.com/kevin-ortega>
:imagesdir: /
:url-prefix:
:url-about: /

You can use link:https://www.ibm.com/cloud/code-engine[IBM Cloud Code Engine] to easily deploy Java applications by using the link:https://openliberty.io/blog/2022/04/01/cloud-native-liberty-buildpack.html[Paketo Buildpack for Liberty] with your container images or source code.

This post provides instructions on how to use the Paketo Buildpack for Liberty with Code Engine. To learn more about using Code Engine, see link:https://cloud.ibm.com/docs/codeengine?topic=codeengine-getting-started[Getting started with IBM Cloud Code Engine].

> With the Paketo Buildpack for Liberty, you can use Open Liberty or WebSphere Liberty to run your application.

Reviewer 2: Remove this sentence.
```suggestion
```

IBM Cloud announced the link:https://cloud.ibm.com/docs/cloud-foundry-public?topic=cloud-foundry-public-deprecation[deprecation] of Cloud Foundry.
Code Engine provides the next level of easy application deployment and mature management of critical business applications in IBM Cloud.  Move your applications from the Cloud Foundry liberty-for-java buildpack to a cloud native solution using Code Engine and the Paketo Buildpack for Liberty.

== What do you need?
* An https://www.ibm.com/cloud[IBM Cloud] Account
* Your application source

To deploy container images, you'll also need:

* The link:https://buildpacks.io/docs/tools/pack/[pack CLI]

> * https://www.docker.com/[Docker]

Reviewer 2: @kevin-ortega, can alternatives like podman be used, or is docker a must?

Reviewer 2: If podman can be used, we should call that out.

* An externally accessible container registry that can host your images.

For details on deploying your application from repository source code from the Code Engine console, see  link:https://cloud.ibm.com/docs/codeengine?topic=codeengine-app-source-code#deploy-app-source-code[Deploying your app from repository source code from the console].

> For details on deploying your application from repository source code with the CLI, see  link:https://cloud.ibm.com/docs/codeengine?topic=codeengine-app-source-code#deploy-app-source-code-cli[Deploying your app from repository source code with the CLI].

Reviewer 2: The links of these sentences are redundant with the part before them (and also there's an extra space after 'see').  This should be restructured to remove the redundancy. perhaps something like 
For details on deploying your application from repository source code using:
* link:https://cloud.ibm.com/docs/codeengine?topic=codeengine-app-source-code#deploy-app-source-code[Code Engine console]
* link:https://cloud.ibm.com/docs/codeengine?topic=codeengine-app-source-code#deploy-app-source-code-cli[CLI]

Similarly, if you're creating your application from source using the Code Engine CLI you need to specify the `port` and build strategy along with the `.buildenv` file present in the application source root directory:

  $ ibmcloud ce app create --name MYAPPNAME --image REGISTRY/NAMESPACE/REPOSITORY --registry-secret SECRET --build-source . --strategy buildpacks --port 9080

```
[finish (prod=)]$ ls -la
total 24
drwxr-xr-x   7 kevin  staff   224 Oct 28 14:54 .
drwxr-xr-x  12 kevin  staff   384 Oct 19 11:07 ..
-rw-r--r--@  1 kevin  staff    26 Nov  4 12:53 .buildenv
-rw-r--r--   1 kevin  staff   790 Oct 19 11:07 Dockerfile
-rw-r--r--@  1 kevin  staff  4066 Oct 19 11:07 pom.xml
drwxr-xr-x   4 kevin  staff   128 Oct 19 11:07 src
drwxr-xr-x  11 kevin  staff   352 Oct 20 08:53 target
```

.To summarize, two important configuration steps are required to use the Paketo Buildpack for Liberty in Code Engine:
. Create a `.buildenv` file that defines the  `BP_JAVA_APP_SERVER=liberty` environment variable.
. Set the listening port to 9080

== Other environment variables you can set in the `.buildenv` file
BP_LIBERTY_INSTALL_TYPE::

> Specifies the  link:https://github.com/paketo-buildpacks/liberty#install-types[Install type] of Liberty. Possible values are `ol` (Open Liberty) or  `wlp` (WebSphere Liberty).  Open Liberty is the default.

Reviewer 2: ```suggestion
Specifies the link:https://github.com/paketo-buildpacks/liberty#install-types[Install type] of Liberty.  Open Liberty (value of `ol`) is the default.
```

BP_LIBERTY_PROFILE::

> Specifies which liberty profile to install. Valid profiles for Open and WebSphere Liberty are documented link:https://github.com/paketo-buildpacks/liberty#profiles[in the buildpacks documentation].

Reviewer 2: ```suggestion
Specifies which liberty profile to install. Valid profiles for Liberty are documented link:https://github.com/paketo-buildpacks/liberty#profiles[in the buildpacks documentation].
```
